include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -350,4 +350,20 @@ def RegResetWithOneReset : Pat<
   (DropWrite $reg, (NodeOp $resetVal, $name, $nameKind, $annotations, $inner_sym)),
   [(OneConstantOp $reset)]>;
 
+//===----------------------------------------------------------------------===//
+// Subaccesses on InvalidValueOp
+//===----------------------------------------------------------------------===//
+
+// subfield(invalidvalue) -> invalidvalue
```

Comment:
This is only true if the subfield is the only use of the invalid value.  Same for the other patterns.
All users of an invalid value must see the same value for it.  Not doing this was the main source of all our original invalid value bugs.